pages/proyectos-de-ley/index.js

```diff
@@ -32,21 +33,73 @@ const useLegislatures = () => {
       fetch(`${process.env.api}legislature`).then(res => res.json()),
   });
   return {
-    ...response,
     isLegislaturesLoading: response.isLoading,
     isLegislaturesSuccess: response.isSuccess,
     legislatures: response.data?.data ?? [],
   };
 };
 
-// TODO: remove placeholders
-const estadoOptions = ['En comisión', 'En comisión2'];
+const useBillStatus = () => {
+  const response = useQuery({
+    queryKey: 'bill-status',
+    queryFn: async () =>
+      fetch(`${process.env.api}bill-status`).then(res => res.json()),
+  });
+  return {
+    isBillStatusLoading: response.isLoading,
+    isBillStatusSuccess: response.isSuccess,
+    billStatus: response.data?.data ?? [],
+  };
+};
+
+const getQueryParamsString = params => {
+  return Object.keys(params)
+    .filter(key => params[key])
+    .map(key => `${key}=${params[key]}`)
+    .join('&');
+};
+
+//TODO: add pagination query
+const useBills = ({ filter, page }) => {
+  const response = useQuery({
+    queryKey: ['bills', filter],
+    queryFn: async () => {
+      const queryParams = getQueryParamsString(filter);
+      const url = `${process.env.api}bill?` + queryParams;
+      return fetch(url).then(res => res.json());
+    },
+  });
+  return {
+    isBillsLoading: response.isLoading,
+    isBillsSuccess: response.isSuccess,
+    bills: response.data?.data ?? [],
+  };
+};
+
+export default function Bills() {
+  const route = useRouter();
+  const { query } = route;
 
-export default function Bills({ bills, metadata }) {
   const { isCommitteesLoading, isCommitteesSuccess, committees } =
     useCommittees();
   const { isLegislaturesLoading, isLegislaturesSuccess, legislatures } =
     useLegislatures();
+  const { isBillStatusLoading, isBillStatusSuccess, billStatus } =
+    useBillStatus();
+  const { isBillsLoading, isBillsSuccess, bills } = useBills({ filter: query });
+
+  const handleChange = e => {
+    const newQuery = {
+      ...query,
+      [e.target.name]: e.target.value,
+    };
+    const validatedQuery = removeEmptyAttrFromObject(newQuery);
```

Comment:
Como mención de repente ni necesitamos darle nombre
```suggestion
    const validatedQuery = omitBy(newQuery, isEmpty);
```
o
```suggestion
    const validatedQuery = removeEmptyPropertiesFromObject(newQuery);
```